### PR TITLE
Refactor importer create at pipeline

### PR DESCRIPTION
**Pull request by azexcy**
Related #22500 

Changes proposed in this pull request:
  - Add Importer connector, prepare for subsequent cdc extensions
  - CDC `proto` updated 

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/spi/importer/connector/ImporterConnector.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * @return connector
+     */
+    Object getConnector();
+    
+    /**
+     * Connector type.
+     *
+     * @return connector type.
+     */
+    String getType();
```
Comment (sandynz) [inline kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/spi/importer/connector/ImporterConnector.java:37]: Is `getType()` required?

File: kernel/data-pipeline/cdc/protocol/src/main/proto/CDCResponseProtocol.proto [hunk 2]
```diff
@@ -51,6 +51,22 @@ message CreateSubscriptionResult {
...
+
+message BigDecimalValue {
+  string value = 1;
+}
+
+message ClobValue {
+  string value = 1;
+}
+
 message FetchRecordResult {
```
Comment (sandynz) [inline kernel/data-pipeline/cdc/protocol/src/main/proto/CDCResponseProtocol.proto:70]: Looks `FETCH_RECORDS` was removed in request protocol, does `FetchRecordResult` need to be updated?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java [hunk 3]
```diff
@@ -79,11 +80,11 @@ public final class DefaultImporter extends AbstractLifecycleExecutor implements
     
     private volatile Statement batchDeleteStatement;
     
-    public DefaultImporter(final ImporterConfiguration importerConfig, final PipelineDataSourceManager dataSourceManager, final PipelineChannel channel,
+    public DefaultImporter(final ImporterConfiguration importerConfig, final ImporterConnector importerConnector, final PipelineChannel channel,
                            final PipelineJobProgressListener jobProgressListener) {
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java:84]: `ImporterConnector` is imported here. Some questions:
1. Is there new type of Importer impl for CDC?
2. If yes, it's better to update related factory `DefaultImporterCreator`. If no, is it enough to reuse `DefaultImporter`?

File: kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java [hunk 4]
```diff
@@ -79,11 +80,11 @@ public final class DefaultImporter extends AbstractLifecycleExecutor implements
...
         this.importerConfig = importerConfig;
         rateLimitAlgorithm = importerConfig.getRateLimitAlgorithm();
-        this.dataSourceManager = dataSourceManager;
+        this.dataSourceManager = (PipelineDataSourceManager) importerConnector.getConnector();
```
Comment (sandynz) [inline kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/importer/DefaultImporter.java:87]: It's better not use type casting here if there's another way

**Review by sandynz — CHANGES_REQUESTED** [review]

File: kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/spi/importer/connector/ImporterConnector.java (hunk 1 above)
Comment (azexcy) [reply kernel/data-pipeline/api/src/main/java/org/apache/shardingsphere/data/pipeline/spi/importer/connector/ImporterConnector.java:37]: Improved, the `getType()` used for load SPI implemention

**Review by sandynz — APPROVED** [review]

**Pull request merged**